tests/test_default_headers.py

```diff
@@ -20,6 +27,41 @@ async def test_default_default_headers():
             assert response.headers["server"] == "uvicorn" and response.headers["date"]
 
 
+@pytest.mark.anyio
+async def test_date_headers_update():
+    config = Config(app=app, loop="asyncio")
+    async with run_server(config):
+        async with httpx.AsyncClient() as client:
+            response = await client.get("http://127.0.0.1:8000")
+            date = response.headers["date"]
+            first_date = dt.datetime.strptime(date, DATE_FORMAT)
+            second_date = first_date
+
+            cancelled = False
+
+            async def sleep_and_cancel(tg: anyio.abc.TaskGroup):
+                nonlocal cancelled
+                await asyncio.sleep(2)
+                cancelled = True  # pragma: no cover
+                await tg.cancel_scope.cancel()  # pragma: no cover
+
+            async def ensure_different_date(tg: anyio.abc.TaskGroup):
+                nonlocal second_date
+                async with httpx.AsyncClient() as client:
+                    while second_date == first_date:
+                        response = await client.get("http://127.0.0.1:8000")
+                        date = response.headers["date"]
+                        second_date = dt.datetime.strptime(date, DATE_FORMAT)
+                tg.cancel_scope.cancel()
+
+            async with anyio.create_task_group() as tg:
+                tg.start_soon(sleep_and_cancel, tg)
+                tg.start_soon(ensure_different_date, tg)
+
+            assert not cancelled
+            assert second_date - first_date == dt.timedelta(seconds=1)
+
```

Comment:
seems convoluted to me but if this works. 
do we know why previous test was failing on ;ac os ?

Reply:
I agree it's convoluted. I can explain the code on the docstring... 

I don't know why MacOS fails. I can try to investigate that before merging it. But the thing is that it failed for timestamp <1 and >1.